petastorm/pytorch.py

```diff
@@ -92,7 +94,39 @@ def decimal_friendly_collate(batch):
         return default_collate(batch)
 
 
-class DataLoader(object):
+_PARALLEL_ITER_ERROR = "You must finish a full pass of Petastorm DataLoader before making another pass from the \
+beginning.If you do need to terminate early and restart from beginning, please re-create the reader and the data \
+loader."
+
+
+class LoaderBase(object):
+
+    def __init__(self):
+        self._in_iter = None
+        self._error = None
+
+    def __iter__(self):
+        if self._error is not None:
+            raise RuntimeError('Cannot start a new iteration because last time iteration failed with error {err}.'
+                               .format(err=repr(self._error)))
+        if self._in_iter is not None and self._in_iter == True:  # noqa: E712
+            raise RuntimeError(_PARALLEL_ITER_ERROR)
+        if self._in_iter is not None:
+            self.reader.reset()
+            logger.warning('Start a new pass of Petastorm DataLoader, reset underlying Petastorm reader to position 0.')
+        self._in_iter = True
+
+        try:
+            for batch in self._iter_impl():
+                yield batch
+        except RuntimeError as e:
```

Comment:
Curious, why did you choose `RuntimeError` and not higher exception class (`Exception`)?